modules/camera/web/face_detection.js

```diff
@@ -0,0 +1,306 @@
+                    if (!isCompletelyInFrame) {
+                        qualityIssues.push('Move face away from edges');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 3. Face Alignment (Roll): Check tilt angle (should be less than 6°)
+                    const leftEyeCenter = {
+                        x: leftEye.reduce((sum, p) => sum + p.x, 0) / leftEye.length,
+                        y: leftEye.reduce((sum, p) => sum + p.y, 0) / leftEye.length
+                    };
+                    const rightEyeCenter = {
+                        x: rightEye.reduce((sum, p) => sum + p.x, 0) / rightEye.length,
+                        y: rightEye.reduce((sum, p) => sum + p.y, 0) / rightEye.length
+                    };
+                    const dx = rightEyeCenter.x - leftEyeCenter.x;
+                    const dy = rightEyeCenter.y - leftEyeCenter.y;
+                    const tiltAngle = Math.abs(Math.atan2(dy, dx) * (180 / Math.PI));
+                    if (tiltAngle > 6) {
+                        qualityIssues.push(`Keep face straight (tilt: ${tiltAngle.toFixed(1)}°)`);
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 4. Face Centering (Horizontal): Check if the nose tip is near the center horizontally
+                    const faceCenterX = box.x + (box.width / 2);
+                    const noseTip = nose[Math.floor(nose.length / 2)]; // Use central nose point
+                    const centeredThreshold = box.width * 0.08; // 8% of face width tolerance
+                    if (Math.abs(noseTip.x - faceCenterX) > centeredThreshold) {
+                        qualityIssues.push('Center face horizontally');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 5. Eye Openness: Use Eye Aspect Ratio (EAR)
+                    const getEAR = (eye) => {
+                        // EAR formula: (||p2 - p6|| + ||p3 - p5||) / (2 * ||p1 - p4||)
+                        const A = distance(eye[1], eye[5]);
+                        const B = distance(eye[2], eye[4]);
+                        const C = distance(eye[0], eye[3]);
+                        return (A + B) / (2.0 * C);
+                    };
+                    const leftEAR = getEAR(leftEye);
+                    const rightEAR = getEAR(rightEye);
+                    const minEAR = 0.25; // Threshold below which eyes are considered closed
+                    if (leftEAR < minEAR || rightEAR < minEAR) {
+                        qualityIssues.push('Eyes appear closed');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 6. Face Size: Check if the face occupies a reasonable portion of the frame
+                    const minFaceWidthRatio = 0.18;
```

Comment:
these should be coming from the app developer as configuration options instead of being hardcoded like this